cognite/seismic/protos/query_service.proto

```diff
@@ -7,157 +7,87 @@ import "cognite/seismic/protos/query_service_messages.proto";
-    **/
-    rpc GetCubeByGeometry (GeometryCubeRequest) returns (stream Trace) {}
-    /**
-    Returns a SEG-Y file.
-    Can retrieve a full file or create a new cropped file filtering on areas of interest only.
-    **/
-    rpc GetSegYFile (SegYQueryRequest) returns (stream SegYQueryResponse) {}
-
-
-    // Depth-indexed area queries
-
-    /**
-    Returns a horizontal slice for a given depth or time and area constrained by a range of inlines and crosslines
-    **/
-    rpc GetTimeSliceByLines (LineTimeSliceQueryRequest) returns (stream SurfacePoint) {}
-    /**
-    Returns a horizontal slice for a given depth or time and area constrained by an arbitrary 2D polygon
-    **/
-    rpc GetTimeSliceByGeometry (GeometryTimeSliceQueryRequest) returns (stream SurfacePoint) {}
-
-    // Volume queries
-    rpc GetVolume(VolumeRequest) returns (stream Trace) {}
+  // Metadata queries
+
+  /**
+  Finds one survey given its name or id. Optionally, lists its associated files.
+  **/
+  rpc GetSurvey (SurveyQueryRequest) returns (GetSurveyResponse) {}
+  /**
+  Lists all surveys owned by this project. Optionally, includes their lists of files.
+  **/
+  rpc ListSurveys (ListSurveysQueryRequest) returns (SurveyWithFilesResponse) {}
+  /**
+  Lists all files available, both owned by the authorized CDF project and shared with it
+  **/
+  rpc ListFiles (google.protobuf.Empty) returns (ListFilesResponse) {}
+  /**
+  Search surveys based on two criteria:
+      Coverage polygon of files in the survey are within an area delimited by a specified polygon
+      Filters on metadata of both the survey and the file
+  Both criteria are optional and can be combined for a more detailed search.
+  **/
+  rpc SearchSurveys (SearchSurveyRequest) returns (SurveyWithFilesResponse) {}
+  /**
+  Returns file metadata given its name or id.
+  **/
+  rpc GetFile (FileQueryRequest) returns (GetFileResponse) {}
+  /**
+  Returns a binary header given its file name or id.
+  **/
+  rpc GetBinaryHeader (HeaderFileQueryRequest) returns (GetBinaryHeaderResponse) {}
+  /**
+  Returns a text header given its file name or id.
+  **/
+  rpc GetTextHeader (HeaderFileQueryRequest) returns (GetTextHeaderResponse) {}
+
+
+  // Geometry and grid queries
+  /**
+  Returns the coverage for a given file identified by its id or name.
+  The coverage is represented by a polygon either in WKT or geojson and represents the area covered by traces
+  in the file.
+  There can be holes in the polygon if traces do not exist in an area inside of it.
+  **/
+  rpc GetFileDataCoverage (FileCoverageRequest) returns (DataCoverageResponse) {}
+  /**
+  Returns the set of valid crossline indices for a specific inline, in a given file
+  **/
+  rpc GetCrosslinesByInline (FileLineQueryRequest) returns (AvailableLines) {}
+  /**
+  Returns the set of valid inline indices for a specific crossline, in a given file
+  **/
+  rpc GetInlinesByCrossline (FileLineQueryRequest) returns (AvailableLines) {}
```

Comment:
Is it the same as GetVolume?

Reply:
Nope, it actually returns the indexes of available ilines for given xlines (and vice versa). No trace data

Reply:
Apparently it's being used for some custom gridding algos outside of CDF.